### ISSUE TITLE
DMD on video

### ISSUE THREAD

**Issue by luangchaungoeuo** [tianweiy/DMD2#54]
I came across your paper, "From Slow Bidirectional to Fast Causal Video Generators," and found it very insightful for applying DMD to video generation. Do you have plan to release the code in the near future? Thank you! @tianweiy

**Comment by tianweiy** [comment]
I’ve recently received a few inquiries, so I’d like to provide a general update:

The original pretrained teacher model and dataset are proprietary to Adobe, which unfortunately means we cannot release them. However, we are committed to providing a reproducible and accessible version. This will include a public dataset, a publicly available teacher model, and the training code required to fine-tune the teacher into a few-step causal student model.

Currently, we are working through the details of this version. For instance, we are evaluating which teacher model to use (e.g., CogVideoX, Mochi, etc.) and which public dataset to select (e.g., OpenVID, PANDA, etc.). Additionally, we need to reimplement the entire codebase from scratch, as our existing results rely on Adobe's proprietary internal codebase.

Given the complexity of these tasks, this effort will take considerable time, and we are unable to provide a specific timeline at the moment. That said, please rest assured that we are doing our best to make this available as soon as possible.

Thank you again for your interest and support!

**Comment by Quick-Eyed-Sky** [comment]
Great!

**Comment by luangchaungoeuo** [comment]
Thanks and looking foward to the public release of the code!

**Comment by pldlgb** [comment]
looking foward to the public release of the code!

**Comment by tianweiy** [comment]
thanks, it will be out in a few weeks. It will include 1. bidirectional DMD 2. causal DMD similar to https://causvid.github.io/ 

please stay tuned.

**Comment by pldlgb** [comment]
Thank you, are there any new updates~ Or any timeline and plans regarding open source~

**Comment by tianweiy** [comment]
I met a few new challenges for causal model. I am still tuning them and probably 1-2 weeks more depending on my progress

**Comment by pldlgb** [comment]
Perhaps you could first open source the bidirectional distillation method, or could you share the data and models you used for distillation? I also encountered some problems when distilling CogvideoX🤣

**Comment by tianweiy** [comment]
bidirectional with cogvideox seems easy. we get it working quickly (we just use cogvideox and prompts from the pixabay dataset. we do backward simulation in dmd2 so that we don't need video data. but using video data should work similarly). causal model is more challenging due to the modulation of text embedding by the timestamp feature (in adaln layer). This adaln layer doesn't work with per-frame timestep ... so finetuning cogvideox to be causal is more challenging. we are trying out some creative solutions which take a bit more time than i expected ...

**Comment by caiqi** [comment]
@tianweiy So for video distillation, the discriminator loss is not used since there is no real video training data? Thank!

**Comment by caiqi** [comment]
Looking forward to the code! My experience is that in image distillation, students could even outperform teachers (might be thanks to the GAN loss), but it's tough to replicate that in video models—probably because handling temporal coherence and motion preservation is way harder than static images. Can't wait to see how you tackle these challenges!
